src/components/block-button.js

```diff
@@ -0,0 +1,31 @@
+window.blockUser = function(id) {
+  NAF.connection.adapter.publisher.handle.sendMessage({ kind: "block", whom: id });
+  NAF.connection.entities.removeEntitiesOfClient(id);
+};
+
+window.unblockUser = function(id) {
+  NAF.connection.adapter.publisher.handle.sendMessage({ kind: "unblock", whom: id });
+  window.setTimeout(() => {
+    // Wait enough time for the unblock to be processed
+    NAF.connection.entities.completeSync(id);
+  }, 1000);
+};
+
+AFRAME.registerComponent("block-button", {
+  init() {
+    this.el.addEventListener("click", this.onClick);
```

Comment:
You need to do this aftaer assigning `this.onClick`, hoisting only applies to varaible declarations.